spec/ble-sparse.test.sh

```diff
@@ -1200,5 +1200,49 @@ a[-1]=1
 ## OK bash STDERR:
 bash: line 3: a[-1]: bad array subscript
 ## END
-## N-I mksh status: 0
+## N-I mksh status: 99
 ## N-I mksh stderr-json: ""
+
+
+#### Initializing indexed array with ([index]=value)
+case $SH in mksh) exit 99;; esac
+declare -a a=([xx]=1 [yy]=2 [zz]=3)
+echo status=$?
+argv.py "${a[@]}"
+## STDOUT:
+status=0
+['3']
+## END
+## N-I mksh status: 99
+## N-I mksh stdout-json: ""
+
+
+#### bash mangles indexed array #1 (keys undergoes arithmetic evaluation)
```

Comment:
Oh I see, why put this test in ble-sparse?   I think it would be better if the 2 tests are next to each other -- indexed array vs. associative array

Reply:
Yeah, I think it would be better to have them in the same place. Where should we put these?

The original location was `spec/assoc.test.sh`, which implies that the test case is intended for  BashAssoc. Then, I moved the test case for BashArray (with the specified index `[i]=v`) to the file for the sparse array, i.e., `spec/sparse.sh`.

Actually, `spec/ble-sparse.test.sh` may be renamed to just `spec/sparse.test.sh`. What do you think?

Reply:
Oh I see, maybe we can have

```
spec/indexed-array  # aka sparse array, maybe we need indexed-array-1 and indexed-array-2, for the ble-sparse
spec/assoc-array

spec/bash-array-literal  # common stuff ?
```

This actually relates to a renaming I want to do

I think we should rename:

- BashArray / BashAssoc types ->  IndexedArray / AssocArray
- internal details: YshArrayLiteral and BashArrayLiteral
  - the BashArrayLiteral is for both IndexedArray / AssocArray

I think this is more consistent with bash, and also solves some of our naming issues ?

I will merge this PR, and then we can talk about it separately

Reply:
#2282